src/typing/flow_js.ml

```diff
@@ -3696,6 +3696,15 @@ let rec __flow cx ((l: Type.t), (u: Type.use_t)) trace =
       ) in
       rec_flow_t cx trace (ArrT (reason, f t, List.map f ts), tout)
 
+    | _, TupleMapT (reason, funt, tout) ->
```

Comment:
Yeah, I think you're right. I was worried about the `VoidT` case being captured by the new rule instead of falling through, but looking at the tests I think it's handled properly - the new error is different but fine. 

Personally I'd ask if you could update (or simply remove) the comments in that `CustomFun` rule which are now out of date, and update the ones in `tests/promises/all` which say "expected array instead of", to reflect the newer more general behavior. Otherwise it looks good to me! @samwgoldman WDYT, good to go mod comments?